### DISCUSSION TITLE
Is it in the roadmap to customize the way the router-cli generates routes?

### DISCUSSION THREAD

**Discussion by ValenCassa** [TanStack/router#789] [General]
Hey! I'd really love the router-cli to generate the routes based on the file structure that Remix uses. Right know I find that it quickly becomes messy, mainly because folders are placed first, so directories and files that are related to each other are not "ordered"

**Comment by tannerlinsley** [comment]
I’m not aware of how it’s different as of today. We support both directory and flat style.

Tanner Linsley

**Reply by ValenCassa** [reply]
In Remix, [you can create a folder per route and/or layout](https://remix.run/docs/en/main/file-conventions/routes#folders-for-organization), so you could have something like:
```
app/
├── routes/
│   ├── app._index/ <--- /app
│   │   ├── route.tsx
│   │   └── stats.tsx
│   ├── app.projects/  <--- /app/projects
│   │   ├── get-projects.server.tsx
│   │   ├── project-buttons.tsx
│   │   ├── project-card.tsx
│   │   └── route.tsx
│   ├── app/ <--- Layout for /app*
│       ├── footer.tsx
│       ├── primary-nav.tsx
│       └── route.tsx
│   
└── root.tsx
```
